src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt

```diff
@@ -0,0 +1,176 @@
+import javax.validation.constraints.NotBlank
+
+@Validated
+@RestController
+class GroupsController(
+  private val groupsService: GroupsService
+) {
+
+  @GetMapping("/groups/{group}")
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_OAUTH_USERS', 'ROLE_AUTH_GROUP_MANAGER')")
+  @Operation(
+    summary = "Group detail.",
+    description = "return Group Details"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Group not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun getGroupDetail(
+    @Parameter(description = "The group code of the group.", required = true)
+    @PathVariable
+    group: String
+  ): GroupDetails {
+    return groupsService.getGroupDetail(group)
+  }
+
+  @PutMapping("/groups/child/{group}")
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_OAUTH_USERS')")
+  @Operation(
+    summary = "Amend child group name.",
+    description = "Amend a Child Group Name"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Child Group not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun amendChildGroupName(
+    @Parameter(description = "The group code of the child group.", required = true)
+    @PathVariable
+    group: String,
+    @Parameter(
+      description = "Details of the child group to be updated.",
+      required = true
+    ) @RequestBody
+    groupAmendment: GroupAmendment
+
+  ) {
+    groupsService.updateChildGroup(group, groupAmendment)
+  }
+}
+
+@Schema(description = "Group Name")
+data class GroupAmendment(
+  @Schema(required = true, description = "Group Name", example = "HDC NPS North East")
+  @field:NotBlank(message = "parent group code must be supplied")
+  val groupName: String
+)
+
+@Schema(description = "User Role")
+data class AuthUserAssignableRole(
+  @Schema(required = true, description = "Role Code", example = "LICENCE_RO")
```

Comment:
I am assuming we will change the name of this class at some point as it should no longer refer to Auth